### DISCUSSION TITLE
Running archinstall from an arch docker image

### DISCUSSION THREAD

**Discussion by l4zy0n3** [archlinux/archinstall#347] [Ideas]
Creating a dockerfile and docker image builds will improve the platform support, as it can then be used from *nix/windows/mac

**Comment by Torxed** [comment]
This would be awesome, I @Didr? ^_^

**Reply by l4zy0n3** [reply]
Dockerfile: https://github.com/l4zy0n3/archinstall/blob/master/Dockerfile
Image: https://hub.docker.com/r/l4zy/archinstall

**Reply by l4zy0n3** [reply]
Created a repo for now, will raise a PR once stable enough

**Reply by l4zy0n3** [reply]
Stuck at this, as docker images don't use systemd, this might require some low level changes
```
File "/usr/lib/python3.9/site-packages/archinstall/lib/general.py", line 259, in run
raise SysCallError(message=f"{self.trace_log.decode('UTF-8')}\n'{self.raw_cmd}' did not exit gracefully (trace log above), exit code: {self.exit_code}", exit_code=self.exit_code)
archinstall.lib.exceptions.SysCallError: System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down

'systemctl show -p SubState --value reflector.service' did not exit gracefully (trace log above), exit code: 256
```

**Reply by Torxed** [reply]
Oh, docker doesn't use systemd? What about EFI?

**Reply by l4zy0n3** [reply]
> Oh, docker doesn't use systemd? What about EFI?

Currently I'm trying to make systemd work in an arch container, it'll require some testing from my end, as for EFI I'm not really sure

**Reply by Torxed** [reply]
> > Oh, docker doesn't use systemd? What about EFI?
> 
> Currently I'm trying to make systemd work in an arch container, it'll require some testing from my end, as for EFI I'm not really sure

We will support non-efi booted machines in v2.2.0, so EFI isn't a huge issue.
The systemd tho is, but if that's a deal breaker there's not many places that needs to be adjusted acutally, so let me know if we should add error-handling where systemd isn't available *(much like we have with offline features to support libraries such as [archoffline](https://github.com/Torxed/archoffline) (not a shameful plug, just an example))*

**Reply by l4zy0n3** [reply]
> > > Oh, docker doesn't use systemd? What about EFI?
> > 
> > Currently I'm trying to make systemd work in an arch container, it'll require some testing from my end, as for EFI I'm not really sure
> 
> We will support non-efi booted machines in v2.2.0, so EFI isn't a huge issue.
> The systemd tho is, but if that's a deal breaker there's not many places that needs to be adjusted acutally, so let me know if we should add error-handling where systemd isn't available *(much like we have with offline features to support libraries such as [archoffline](https://github.com/Torxed/archoffline) (not a shameful plug, just an example))*

That seems like a pretty good idea for a fallback, I used to use Artix ( with runit as init system), such distros will also get support,
For now I have gotten systemd to work but it's giving me a login prompt for a regular user( instead of root) inside the container, trying to find a workaround for that

**Reply by Torxed** [reply]
I'll create an issue, for anyone reading this thread: Fear not, this is outside of the guided installer. These offline and non-systemd changes are in the API part of archinstall and won't affect normal users experiences with the guided installer.
